src/Model.php

```diff
@@ -1225,14 +1234,45 @@ public function addCondition($field, $operator = null, $value = null)
             */
         }
 
-        $f = is_string($field) ? $this->getField($field) : ($field instanceof Field ? $field : false);
-        if ($f) {
+        if (is_string($field)) {
+            // shorthand for adding conditions on references
+            // use chained reference names separated by "/"
+            if (stripos($field, '/') !== false) {
+                $references = explode('/', $field);
+
+                $field = array_pop($references);
+
+                $model = $this;
+                foreach ($references as $link) {
+                    $model = $model->refLink($link);
+                }
+
+                $args = func_get_args();
+
+                // '#' will apply condition directly on the record count (has # referenced records)
+                // otherwise applying condition on the referenced model field (has referenced records where)
+                if ($field !== '#') {
+                    $model->addCondition(...$args);
```

Comment:
Shouldn't we use `($field, $operator, $value)` here instead? Otherwise it will pass field as `company/country` for example, but should pass only extracted field name `country`.

For now it's OK, but maybe think about how we could use `SELECT * FROM t WHERE EXISTS (...)` syntax in this case (for SQL). This would be better for performance than counting records.

Reply:
`func_get_args` takes the last value of the argument variable. So in this case we change `$field` to the field name of the reference a few lines above.
`func_get_args` is used intentionally as it takes only the number of arguments passed onto the method (after last changes to the variables). This is important as `$operator` can also be the `$value` so if you pass `($field, $operator, $value)` you always pass on 3 arguments and the shorthand for `=` operator will not work.
What I noticed also is that `func_get_args` takes only the number of arguments originally passed onto the method. So if you have only `($field, $operator)` and you set `$value` to something it will not get included in the `func_get_args` array. That is why I used `$args` variable.
I was also considering `EXISTS` syntax as I know it is much faster. Will think of some field syntax for that: maybe using '?' for exists and '!' does not exist?

Reply:
Wow. I didn't know that. Good catch !